### PR TITLE
Ansible Collection Migration

### PR DESCRIPTION
**Pull request by rambleraptor**
I think it's time to get rid of our Ansible fork!

This PR:
* Changes the submodule
* Removes the unit testing from the Magician (collections don't support linting tests yet, so we're going to test in prod for a hot minute. I believe that testing shouldn't occur in the Magician anyways)
* Changes the file paths
* Changes the Magician
* Makes changes to the release pipeline (collection -> core) that may be unnecessary (I don't know how often I'll be committing to core in this interim period) and might not work properly

# Release Note for Downstream PRs (will be copied)
```releasenote

```

### CODE REVIEW

File: .ci/magic-modules/release-ansible.sh [hunk 1]
```diff
@@ -49,6 +49,17 @@ git remote add upstream git@github.com:ansible/ansible.git
 git remote add magician git@github.com:modular-magician/ansible.git
 echo "Remotes setup properly"
 
+# Create a commit on our fork using the code from the collection.
```
Comment (danawillow) [inline .ci/magic-modules/release-ansible.sh]: can you elaborate on why?

File: provider/ansible.rb [hunk 2]
```diff
@@ -257,7 +257,7 @@ def generate_resource_tests(data)
 
         name = module_name(data.object)
         path = File.join(target_folder,
-                         "test/integration/targets/#{name}/tasks/main.yml")
+                         "tests/integration/#{name}/tasks/main.yml")
```
Comment (danawillow) [inline provider/ansible.rb]: https://github.com/ansible/ansible_collections_google is showing me a directory structure of `tests/integration/targets/[...]`, is that supposed to match these?

Comment (rambleraptor) [reply provider/ansible.rb]: you beat me to the punch! I was playing around with the proper structure on the repo and had to backport that change (back).

File: .ci/magic-modules/release-ansible.sh (hunk 1 above)
Comment (rambleraptor) [reply .ci/magic-modules/release-ansible.sh]: done!

File: .ci/ci.yml.tmpl [hunk 3]
```diff
@@ -268,29 +268,6 @@ jobs:
                 get_params:
                   skip_clone: true
 
-    - name: ansible-test
```
Comment (rambleraptor) [inline .ci/ci.yml.tmpl]: This is a temporary measure until everything works nicely on with their test runner. I'll file an issue to reintroduce unit testing back into the Magician

**Comment by rambleraptor** [comment]
@danawillow ptal when you can!

**Pull request merged**